src/dimension_range.c

```diff
@@ -0,0 +1,1127 @@
+	range_space->num_range_cols = 0;
+
+	/* Perform an index scan on hypertable_id, invalid chunk_id. */
+	ScanKeyInit(
+		&scankey[0],
+		Anum_dimension_range_ht_id_chunk_id_column_name_range_start_range_end_idx_hypertable_id,
+		BTEqualStrategyNumber,
+		F_INT4EQ,
+		Int32GetDatum(hypertable_id));
+	ScanKeyInit(&scankey[1],
+				Anum_dimension_range_ht_id_chunk_id_column_name_range_start_range_end_idx_chunk_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(INVALID_CHUNK_ID));
+
+	column_dimension_range_scan_internal(
+		scankey,
+		2,
+		dimension_range_tuple_found,
+		range_space,
+		0,
+		DIMENSION_RANGE_HT_ID_CHUNK_ID_COLUMN_NAME_RANGE_START_RANGE_END_IDX,
+		AccessShareLock,
+		mctx);
+
+	if (range_space->num_range_cols == 0)
+	{
+		pfree(range_space);
+		return NULL;
+	}
+
+	return range_space;
+}
+
+static ScanTupleResult
+form_range_tuple_found(TupleInfo *ti, void *data)
+{
+	Form_dimension_range rg = data;
+	bool should_free;
+	HeapTuple tuple = ts_scanner_fetch_heap_tuple(ti, false, &should_free);
+
+	memcpy(rg, GETSTRUCT(tuple), sizeof(FormData_dimension_range));
+
+	if (should_free)
+		heap_freetuple(tuple);
+
+	return SCAN_DONE;
+}
+
+Form_dimension_range
+ts_column_dimension_range_lookup(int32 hypertable_id, int32 chunk_id, const char *col_name)
+{
+	ScanKeyData scankey[3];
+	Form_dimension_range form_range = palloc0(sizeof(FormData_dimension_range));
+	form_range->chunk_id = INVALID_CHUNK_ID; /* for clarity */
+
+	/* Perform an index scan on hypertable_id, chunk_id, col_name. */
+	ScanKeyInit(
+		&scankey[0],
+		Anum_dimension_range_ht_id_chunk_id_column_name_range_start_range_end_idx_hypertable_id,
+		BTEqualStrategyNumber,
+		F_INT4EQ,
+		Int32GetDatum(hypertable_id));
+	ScanKeyInit(&scankey[1],
+				Anum_dimension_range_ht_id_chunk_id_column_name_range_start_range_end_idx_chunk_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(chunk_id));
+	ScanKeyInit(
+		&scankey[2],
+		Anum_dimension_range_ht_id_chunk_id_column_name_range_start_range_end_idx_column_name,
+		BTEqualStrategyNumber,
+		F_NAMEEQ,
+		CStringGetDatum((col_name)));
+
+	column_dimension_range_scan_internal(
+		scankey,
+		3,
+		form_range_tuple_found,
+		form_range,
+		1,
+		DIMENSION_RANGE_HT_ID_CHUNK_ID_COLUMN_NAME_RANGE_START_RANGE_END_IDX,
+		AccessShareLock,
+		CurrentMemoryContext);
+
+	if (strlen(NameStr(form_range->column_name)) == 0)
+	{
+		pfree(form_range);
+		return NULL;
+	}
+
+	return form_range;
+}
+
+/*
+ * Update column dimension ranges in the catalog for the
+ * provided chunk (it's assumed that the chunk is locked
+ * appropriately).
+ *
+ * Calculate actual ranges for the given chunk for the columns
+ * insert these entries. This allows for the
+ * chunk to be picked up when queries use these columns in
+ * WHERE clauses with these ranges.
+ *
+ * Returns the number of column entries that have been added or
+ * updated.
+ */
+int
+ts_column_dimension_range_calculate(const Hypertable *ht, Chunk *chunk)
```

Comment:
Consider using const for chunk as well. It looks like possible.

```suggestion
ts_column_dimension_range_calculate(const Hypertable *ht, const Chunk *chunk)
```

Reply:
done.